WcaOnRails/app/models/competition.rb

```diff
@@ -380,10 +383,30 @@ def warnings_for(user)
         warnings[:results] = I18n.t('competitions.messages.results_not_posted')
       end
     end
+    if reg_warnings.any?
+      warnings = reg_warnings.merge(warnings)
+    end
 
     warnings
   end
 
+  def reg_warnings
+    warnings = {}
+    if registration_range_specified? && !registration_past?
+      if self.announced?
+        if (self.registration_open - self.announced_at) < REGISTRATION_OPENING_EARLIEST
+          warnings[:regearly] = I18n.t('competitions.messages.reg_opens_too_early')
+        end
+      else
+        if (self.registration_open - Time.now.utc) < REGISTRATION_OPENING_EARLIEST
+          warnings[:regearly] = I18n.t('competitions.messages.reg_opens_too_early')
+        end
+      end
+
+      warnings
```

Comment:
Ah, I've just noticed this is still in the if block, but should be outside, at the end of this method (so we always return a map).

Reply:
Nice, I corrected it accordingly and it's no longer failing! :D